app/src/main/java/org/astraea/producer/Builder.java

```diff
@@ -98,28 +119,56 @@
             return this;
           }
 
+          @Override
+          public Sender<Key, Value> transaction() {
```

Comment:
Exposing this method causes two ways to execute transaction. I don't think that is a good pattern. Alternative is to change it to private method.

Reply:
Yes, we should provide only one way to call transactional send.

But, this method can not be changed to private method because of assigning "weaker access privileges".

Is there any way to let the method private?

Reply:
It seems to me that users should NOT use `Sender` to send transaction. Instead, they should use `transaction` method. Hence, we can remove this new API from `Sender` interface so as to make this one be private.

Reply:
By doing so, the producer can not access `transaction` method either.

I do transactional send setting in producer's method.

Reply:
Using casting to convert the interface to real type. We should have single Sender impl in this project.